synthtool/languages/java.py

```diff
@@ -92,9 +91,32 @@ def _download_formatter(version: str, dest: Path) -> None:
         fh.write(response.content)
 
 
+HEADER_REGEX = re.compile("\\* Copyright \\d{4} Google LLC")
+
+
+def _file_has_header(path: Path) -> bool:
+    """Return true if the file already contains a license header."""
+    with open(path, "rt") as fp:
+        for line in fp:
+            if HEADER_REGEX.search(line):
+                return True
+    return False
+
+
+def _filter_no_header(paths: Iterable[Path]) -> Iterable[Path]:
+    """Return a subset of files that do not already have a header."""
+    for path in paths:
+        anchor = Path(path.anchor)
+        remainder = str(path.relative_to(path.anchor))
+        for file in anchor.glob(remainder):
+            if not _file_has_header(file):
+                yield file
+
+
 def fix_proto_headers(proto_root: Path) -> None:
+    """Helper to ensure that generated proto classes have appropriate license headers."""
```

Comment:
In the doc string, please describe what the function does.  What actions does it take to fix the proto headers?

Reply:
Done